### PR TITLE
feat(ws): introduce message parser for ws adapter

### PR DESCRIPTION
**Pull request by CodyTseng**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #14027 

## What is the new behavior?

Add a `setMessagePreprocessor` method to the `wsAdapter` to customize how messages are preprocessed.

```ts
const wsAdapter = new WsAdapter(app);

wsAdapter.setMessagePreprocessor((message: any) => {
  const [event, ...data] = message;
  return { event, data };
});
```

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: packages/platform-ws/adapters/ws-adapter.ts [hunk 1]
```diff
@@ -138,7 +142,10 @@ export class WsAdapter extends AbstractWsAdapter {
     transform: (data: any) => Observable<any>,
   ): Observable<any> {
     try {
-      const message = JSON.parse(buffer.data);
+      const message = this.messagePreprocessor(JSON.parse(buffer.data));
```
Comment (kamilmysliwiec) [inline packages/platform-ws/adapters/ws-adapter.ts:145]: I wonder if we should move the "JSON.parse" call to the preprocessor, and just call it parser instead?

Comment (CodyTseng) [reply packages/platform-ws/adapters/ws-adapter.ts:145]: It's a great idea. 

Additionally, should we consider supporting configuration in the constructor, like this:

```typescript
new WsAdapter(app, {
  messageParser: (data) => ...
})
```

Comment (kamilmysliwiec) [reply packages/platform-ws/adapters/ws-adapter.ts:145]: sure, sounds good!

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**